core/templates/dev/head/collection_editor/editor_tab/CollectionLinearizerServiceSpec.js

```diff
@@ -0,0 +1,204 @@
+        secondCollectionNodeBackendObject);
+
+    var thirdCollectionNodeBackendObject = {
+      exploration_id: 'exp_id2',
+      prerequisite_skills: ['exp title1'],
+      acquired_skills: ['exp title2'],
+      exploration_summary: {
+        title: 'exp title2',
+        category: 'exp category',
+        objective: 'exp objective'
+      }
+    };
+    thirdCollectionNode = CollectionNodeObjectFactory.create(
+        thirdCollectionNodeBackendObject);
+  }));
+
+  var _createLinearCollection = function() {
+    var collection = CollectionObjectFactory.createEmptyCollection();
+    collection.addCollectionNode(firstCollectionNode);
+    collection.addCollectionNode(secondCollectionNode);
+    collection.addCollectionNode(thirdCollectionNode);
+    return collection;
+  };
+
+  describe('Collection linearizer node removal function', function() {
+    it('should correctly remove a node from a collection with a single node',
+        function() {
+          var collection = CollectionObjectFactory.createEmptyCollection();
+          collection.addCollectionNode(firstCollectionNode);
+          expect(collection.containsCollectionNode('exp_id0')).toBe(true);
+          CollectionLinearizerService.removeCollectionNode(
+              collection, 'exp_id0');
+          expect(collection.containsCollectionNode('exp_id0')).toBe(false);
+          expect(
+              CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+                  collection)).toEqual([]);
+        });
+
+    it('should correctly remove the first node from a collection', function() {
+      var collection = _createLinearCollection();
+      CollectionLinearizerService.removeCollectionNode(collection, 'exp_id0');
+      expect(collection.containsCollectionNode('exp_id0')).toBe(false);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([secondCollectionNode, thirdCollectionNode]);
+    });
+
+    it('should correctly remove the last node from a collection', function() {
+      var collection = _createLinearCollection();
+      CollectionLinearizerService.removeCollectionNode(collection, 'exp_id2');
+      expect(collection.containsCollectionNode('exp_id2')).toBe(false);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([firstCollectionNode, secondCollectionNode]);
+    });
+
+    it('should correctly remove a middle node from a collection', function() {
+      var collection = _createLinearCollection();
+      CollectionLinearizerService.removeCollectionNode(collection, 'exp_id1');
+      expect(collection.containsCollectionNode('exp_id1')).toBe(false);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([firstCollectionNode, thirdCollectionNode]);
+    });
+  });
+
+  describe('Collection linearizer node append function', function() {
```

Comment:
Also, these tests should ensure collection node prerequisite and acquired skills are properly setup.